code-experiments/build/java/CocoJNI.c

```diff
@@ -298,16 +299,16 @@ JNIEXPORT jdoubleArray JNICALL Java_CocoJNI_cocoEvaluateConstraint
   }
 
   problem = (coco_problem_t *) jproblem_pointer;
-  number_of_objectives = (int) coco_problem_get_number_of_objectives(problem);
+  number_of_constraints = (int) coco_problem_get_number_of_constraints(problem);
 
   /* Call coco_evaluate_constraint */
   x = (*jenv)->GetDoubleArrayElements(jenv, jx, NULL);
-  y = coco_allocate_vector(number_of_objectives);
+  y = coco_allocate_vector(number_of_constraints);
   coco_evaluate_constraint(problem, x, y);
 
   /* Prepare the return value */
-  jy = (*jenv)->NewDoubleArray(jenv, number_of_objectives);
-  (*jenv)->SetDoubleArrayRegion(jenv, jy, 0, number_of_objectives, y);
+  jy = (*jenv)->NewDoubleArray(jenv, number_of_constraints);
+  (*jenv)->SetDoubleArrayRegion(jenv, jy, 0, number_of_constraints, y);
 
```

Comment:
sorry, I accidentally removed the comment here. @brockho was wondering why it was like this before, while it seems OK now.